packages/api/src/utils/billing-types.ts

```diff
@@ -91,19 +91,27 @@ export interface BillingEnv {
   billing: BillingService
   customers: CustomersService
   subscriptions: SubscriptionsService
+  termsOfService: TermsOfServiceService
 }
 
 export type PaymentSettings = {
   paymentMethod: null | PaymentMethod
   subscription: W3PlatformSubscription
+  agreement?: Agreement
 }
 
 export interface UserCustomerService {
   getUserCustomer: (userId: string) => Promise<null|{ id: string }>
   upsertUserCustomer: (userId: string, customerId: string) => Promise<void>
 }
 
+export interface TermsOfServiceService {
+  createUserTosAgreement: (userId: string, agreement: Agreement) => Promise<void>
```

Comment:
where I'm coming from is it may not be long before there are agreements like 'privacy policy' or something that might not be exactly 'terms of service'. afaict they are similar but have important legal distinctions. But still would be nice to be able to share the agreement audit log across both. (which we can totally do without renaming anything, which is something I like about how you've already implemented it!)

Reply:
This makes sense.  Updating.

However I want to note that we currently have a `terms_of_service` table (rather than a generic `agreements` table).  It makes sense to me to model agreement logging in separate tables, since it seems unlikely we'd want to query for all agreements for a given user (regardless of their type), but I wanted to call this out in case you had other thoughts.  _Should_ we change the `terms_of_service` table to `agreements` and put _all_ agreement logging in that one table?